src/test/java/net/snowflake/client/jdbc/PreparedStatement1IT.java

```diff
@@ -790,4 +791,46 @@ public void manualTestForPreparedStatementLogging() throws SQLException {
       }
     }
   }
+
+  /**
+   * This test shows how to use batch requests with variant columns. Syntax <code>
+   * insert into ... Select column1, ... from values (?,...)</code> is crucial there.
+   * CLIENT_STAGE_ARRAY_BINDING_THRESHOLD parameter is used on session to set the minimal number of
+   * bind parameters to create stage file with provided parameters.
+   */
+  @Test
+  public void shouldUseVariantBindParametersSNOW1531307() throws SQLException {
+    try (Connection con = getConnection();
+        Statement stmt = con.createStatement()) {
+      String tableName = "SNOW1531307";
+      stmt.execute(
+          "CREATE OR REPLACE TABLE " + tableName + "(id text, json1 variant, json2 variant)");
+      executePreparedStatementWithBatches(con, tableName, 20);
+      try {
+        stmt.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 10");
+        executePreparedStatementWithBatches(con, tableName, 20);
+      } finally {
+        // we are planning to reuse connection so let's cleanup
+        stmt.execute("ALTER SESSION UNSET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD");
+      }
+    }
+  }
+
+  private static void executePreparedStatementWithBatches(
```

Comment:
Do we need this funtion? Maybe we can inline it?

Reply:
I prefer to keep it since the code is called twice - just rename to match with the specific test